plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java

```diff
@@ -27,327 +27,289 @@
     }
-  }
-
-  Structure getNodeStructure(String dbName) throws GroundException {
-    try {
-      Node node = this.getNode(dbName);
-      return groundReadWrite.getStructureFactory().retrieveFromDatabase(dbName);
-    } catch (GroundException e) {
-      LOG.error("Unable to fetch database node structure");
-      throw e;
+
+    Structure getEdgeStructure(NodeVersion nodeVersion) throws GroundException {
+        try {
+            LOG.debug("Fetching database table edge structure: " + nodeVersion.getNodeId());
+            Edge edge = this.getEdge(nodeVersion);
+            return groundReadWrite.getStructure(edge.getName());
+        } catch (GroundException e) {
+            LOG.debug("Not found - database table edge structure: " + nodeVersion.getNodeId());
+            throw new GroundException(e);
+        }
     }
-  }
-
-  Edge getEdge(NodeVersion nodeVersion) throws GroundException {
-    String edgeId = "" + nodeVersion.getNodeId();
-    try {
-      LOG.debug("Fetching database table edge: " + edgeId);
-      return groundReadWrite.getEdgeFactory().retrieveFromDatabase(edgeId);
-    } catch (GroundException e) {
-      LOG.debug("Not found - Creating database table edge: " + edgeId);
-      Edge edge = groundReadWrite.getEdgeFactory().create(edgeId);
-      Structure edgeStruct = groundReadWrite.getStructureFactory().create(edge.getName());
-      return edge;
+
+    Database getDatabase(String dbName) throws GroundException {
+        List<String> versions = groundReadWrite.getLatestVersions(dbName);
+        if (versions.isEmpty()) {
+            throw new GroundException("Database node not found: " + dbName);
+        }
+        NodeVersion latestVersion = groundReadWrite.getNodeVersion(versions.get(0));
+        Map<String, Tag> dbTag = latestVersion.getTags();
+        return JsonUtil.fromJSON((String) dbTag.get(dbName).getValue(), Database.class);
     }
-  }
-
-  Structure getEdgeStructure(NodeVersion nodeVersion) throws GroundException {
-    try {
-      LOG.debug("Fetching database table edge structure: " + nodeVersion.getNodeId());
-      Edge edge = this.getEdge(nodeVersion);
-      return groundReadWrite.getStructureFactory().retrieveFromDatabase(edge.getName());
-    } catch (GroundException e) {
-      LOG.debug("Not found - database table edge structure: " + nodeVersion.getNodeId());
-      throw e;
+
+    NodeVersion createDatabase(Database db) throws InvalidObjectException, MetaException {
+        if (db == null) {
+            throw new InvalidObjectException("Database object passed is null");
+        }
+        try {
+            String dbName = db.getName();
+            Node dbNode = this.getNode(dbName);
+            Structure dbStruct = groundReadWrite.createStructure(dbName);
+            LOG.debug("Node and Structure {}, {}", dbNode.getId(), dbStruct.getId());
+            Map<String, GroundType> structVersionAttribs = new HashMap<>();
+            structVersionAttribs.put(dbName, GroundType.STRING);
```

Comment:
You're creating a structure version here. A structure is meant to be like a struct in C where you specify a set of fields that are enforced when you create a node version with that structure version. You're creating a structure where one of the required fields is whatever is stored in the `dbName` variable. e.g., if the string "Ground" was the name of the database, then you would require a Tag with key "Ground". Based on my understanding, it seemed like we would want the key to actually be "dbName" and the value to be "Ground".